src/app/components/keywords/index.tsx

```diff
@@ -61,25 +67,35 @@ export const KeywordsEditor: React.FC<KeywordsEditorProps> = (props) => {
     }
   }, [keywordsDomContainer]);
 
-  const renderKeywords = (keywords: EditorState[]) => {
-    return keywords.map((keywordEditorState, index) => {
+  const renderKeywords = (keywords: Keyword[]): React.ReactNode[] => {
+    return keywords.map((keyword, index) => {
       return (
-        <Keyword
+        <KeywordSection
           key={index}
           isActive={isActiveKeyword(name, index, activeKeywordPath)}
-          onChange={onChange.bind(null, name, index)}
-          editorState={keywordEditorState}
-          onDelete={onDelete.bind(null, name, index)}
-          onFocus={handleFocus.bind(null, index)}
+          onChange={onChange.bind(null, name)}
+          keyword={keyword}
+          onDelete={onDelete.bind(null, name)}
+          onFocus={handleFocus.bind(null)}
           onBlur={onBlur}
         />
       );
     });
   };
 
-  const handleNewKeywordFocus = useCallback(() => {
-    onFocus(name, undefined, true);
-  }, [name, onFocus]);
+  const handleNewKeywordFocus = useCallback(
+    (state: EditorState) => {
+      onFocus(state, `keywordGroups.${name}.keywords.newKeyword.content`);
+    },
+    [name, onFocus]
+  );
+
+  const handleAddKeyword = useCallback(
+    (editorState: EditorState) => {
+      onAdd(name, { id: newKeyword.id, content: editorState });
```

Comment:
If you just hit return in the keywords dialog, it adds an empty keyword. Not sure if this is desired behaviour or not, but I think that there should be a check to make sure the content is 1 or more in length maybe before adding the keyword. We can discuss with Production and Design during the next review meeting.

Reply:
this behaviour is also present in staging but it's a very straightforward fix so I'll tackle it here